### ISSUE TITLE
Pylint MergedConfigParser error when pylint is not even used

### ISSUE THREAD

**Issue by kwint** [flakeheaven/flakeheaven#128]
* [ ] Same issue occurs when reproducing the MWE below: new virtualenv, clean folder, minimal additional libs, etc.
* [ x] Same issue occurs when running without cache (eg by setting the env var `FLAKEHEAVEN_CACHE_TIMEOUT=0`)

**Bug Description**
I tried flakeheaven a few months ago and discarded it because if this error I'm having. Now my colleagues dont have the same issue, and we have moved most of our repos to flakeheaven now.

When I run `flakeheaven lint` or `flakeheaven plugins` I get the error below.

I have no clue what is going on here. We dont use `pylint` anywhere, I cannot find any reference to pylint in the code or config files.

the output of `flakeheaven --version` is: 
```
FlakeHeaven 1.0.2
Flake8    4.0.1
```

I use poetry and have the exact same output when I run it with `poetry run flakeheaven`. `which flakeheaven` shows the proper path to my python env. Do you have any clue what is going wrong here? 

btw, the Discord link when creating a new issue is broken

```
Traceback (most recent call last):
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/plugins/manager.py", line 161, in load_plugin
    self._load()
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/plugins/manager.py", line 138, in _load
    self._plugin = self.entry_point.load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/__init__.py", line 5, in <module>
    from ._cli import entrypoint, flake8_entrypoint
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/_cli.py", line 9, in <module>
    from .commands import COMMANDS
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/commands/__init__.py", line 5, in <module>
    from ._baseline import baseline_command
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/commands/_baseline.py", line 6, in <module>
    from .._patched import FlakeHellApplication
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/_patched/__init__.py", line 2, in <module>
    from ._app import FlakeHellApplication
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakehell/_patched/_app.py", line 10, in <module>
    from flake8.options.config import MergedConfigParser, get_local_plugins
ImportError: cannot import name 'MergedConfigParser' from 'flake8.options.config' (/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/options/config.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/bin/flakeheaven", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/_cli.py", line 40, in entrypoint
    exit_code, msg = main(argv)
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/_cli.py", line 32, in main
    return COMMANDS[command_name](argv=argv[1:])
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/commands/_missed.py", line 18, in missed_command
    installed_plugins = sorted(get_installed(app=app), key=lambda p: p['name'])
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/_logic/_discover.py", line 30, in get_installed
    app.initialize([])
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/main/application.py", line 343, in initialize
    self.find_plugins(config_finder)
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/_patched/_app.py", line 173, in find_plugins
    self.check_plugins.load_plugins()
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/plugins/manager.py", line 422, in load_plugins
    plugins = list(self.manager.map(load_plugin))
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flakeheaven/_patched/_plugins.py", line 51, in map
    yield func(plugin, *args, **kwargs)
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/plugins/manager.py", line 420, in load_plugin
    return plugin.load_plugin()
  File "/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/plugins/manager.py", line 168, in load_plugin
    raise failed_to_load
flake8.exceptions.FailedToLoadPlugin: Flake8 failed to load plugin "pylint" due to cannot import name 'MergedConfigParser' from 'flake8.options.config' (/home/quint/.cache/pypoetry/virtualenvs/metal-core-aCGQNnAQ-py3.10/lib/python3.10/site-packages/flake8/options/config.py).
```

**Comment by kwint** [comment]
Pretty sure I did this before, but I completely removed my poetry env and reinstalled it. Issue is gone now. Feel free to close, although might be good to fix this properly still?

**Comment by pwoolvett** [comment]
hi @kwint . Good to know its been solved.

We've made lots of changes and fixes since `1.0.2` - you can look at them [here](https://github.com/flakeheaven/flakeheaven/blob/main/CHANGELOG.md).

I've updated the discord link to include a permanent ref.

**Comment by javivdm** [comment]
Hi, I am getting the same error.

My versions are:
```
FlakeHeaven 3.0.0
Flake8    4.0.1
```

And this the stack trace:
```
Traceback (most recent call last):
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/plugins/manager.py", line 161, in load_plugin
    self._load()
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/plugins/manager.py", line 138, in _load
    self._plugin = self.entry_point.load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/__init__.py", line 5, in <module>
    from ._cli import entrypoint, flake8_entrypoint
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/_cli.py", line 9, in <module>
    from .commands import COMMANDS
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/commands/__init__.py", line 5, in <module>
    from ._baseline import baseline_command
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/commands/_baseline.py", line 6, in <module>
    from .._patched import FlakeHellApplication
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/_patched/__init__.py", line 2, in <module>
    from ._app import FlakeHellApplication
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakehell/_patched/_app.py", line 10, in <module>
    from flake8.options.config import MergedConfigParser, get_local_plugins
ImportError: cannot import name 'MergedConfigParser' from 'flake8.options.config' (/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/options/config.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/bin/flakeheaven", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakeheaven/_cli.py", line 40, in entrypoint
    exit_code, msg = main(argv)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakeheaven/_cli.py", line 32, in main
    return COMMANDS[command_name](argv=argv[1:])
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakeheaven/commands/_lint.py", line 12, in lint_command
    app.run(argv)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/main/application.py", line 375, in run
    self._run(argv)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/main/application.py", line 363, in _run
    self.initialize(argv)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/main/application.py", line 343, in initialize
    self.find_plugins(config_finder)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakeheaven/_patched/_app.py", line 228, in find_plugins
    self.check_plugins.load_plugins()
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/plugins/manager.py", line 422, in load_plugins
    plugins = list(self.manager.map(load_plugin))
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flakeheaven/_patched/_plugins.py", line 51, in map
    yield func(plugin, *args, **kwargs)
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/plugins/manager.py", line 420, in load_plugin
    return plugin.load_plugin()
  File "/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/plugins/manager.py", line 168, in load_plugin
    raise failed_to_load
flake8.exceptions.FailedToLoadPlugin: Flake8 failed to load plugin "pylint" due to cannot import name 'MergedConfigParser' from 'flake8.options.config' (/home/javi/.cache/pypoetry/virtualenvs/price-tracker-QZj4_4Z0-py3.8/lib/python3.8/site-packages/flake8/options/config.py).
```

**Comment by gotofritz** [comment]
I got it too with poetry. I also ended up uninstalling then re-installing poetry to make it work.
Something is up, but I think it's got more to do with Poetry than flakeheaven.
Specifically, Poetry and VSCode, and navigating different folders with different Poetry shells from the same terminal.
But am unable to test investigate, if I ever do I'll post an update in case someone else comes across this
